catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -314,17 +278,89 @@ protected Object executeWithSubject() throws Exception {
         return null;
     }
 
+    private File getInputFile() {
+        final File inputFile = new File(filePath);
+
+        SecurityLogger.audit("Called catalog:ingest command with path : {}", filePath);
+
+        if (!inputFile.exists()) {
+            printErrorMessage(String.format("File or directory [%s] must exist.", filePath));
+            console.println("If the file does indeed exist, try putting the path in quotes.");
+            return null;
+        }
+
+        if (includeContent && !FilenameUtils.isExtension(filePath, "zip")) {
+            console.printf("File %s must be a zip file.", filePath);
+            return null;
+        }
+
+        if (deprecatedBatchSize != DEFAULT_BATCH_SIZE) {
+            // user specified the old style batch size, so use that
+            printErrorMessage(
+                    "Batch size positional argument is DEPRECATED, please use --batchsize option instead.");
+            batchSize = deprecatedBatchSize;
+        }
+
+        if (batchSize <= 0) {
+            printErrorMessage(String.format(
+                    "A batch size of [%d] was supplied. Batch size must be greater than 0.",
+                    batchSize));
+            return null;
+        }
+
+        if (StringUtils.isNotEmpty(failedDir)) {
+            failedIngestDirectory = new File(failedDir);
+            if (!verifyFailedIngestDirectory()) {
+                return null;
+            }
+
+            // Batch size is always set to one when using an Ingest Failure Directory. If a batch
+            // size is specified by the user, issue a warning stating that a batch size of one will
+            // be used.
+            if (batchSize != DEFAULT_BATCH_SIZE) {
+                console.printf(
+                        "WARNING: An ingest failure directory was supplied in addition to a batch "
+                                + "size of %d. When using an ingest failure directory, the batch "
+                                + "size must be 1. Setting batch size to 1.%n",
+                        batchSize);
+            }
+
+            batchSize = 1;
+        }
+
+        if (!SERIALIZED_OBJECT_ID.matches(transformerId)) {
+            transformer = getTransformer();
+            if (transformer == null) {
+                console.println(transformerId + " is an invalid input transformer.");
+                return null;
+            }
+        }
+        return inputFile;
+    }
+
+    private int totalFileCount(File inputFile) throws IOException {
+        if (inputFile.isDirectory()) {
+            int cnt = 0;
```

Comment:
I would recommend renaming to something more descriptive like `currentDirectoryFileCount`.